### DISCUSSION TITLE
Is it possible to add these Labor Plus versions

### DISCUSSION THREAD

**Discussion by schluckspecht** [Freetz-NG/freetz-ng#909] [Allgemein] [answered]
Hi, in Fritz OS 7.57 there are big annoying problems with Voip and Volte Wifi. But AVM did not fix it in a new release,  but in a FRITZ.Box_7490-07.57-109066-LabPLUS.image. Is it possible to make this file selectable as a basic image for Freetz-ng? Obviously these Plus images are bugfix images, not "Labor" files where new features are tested.

**Comment by aus-sua** [comment]
"früher" ging so etwas relativ einfach per menueconfig per:
Download options  --->" "
Weil viele damit aber nur "rumgestümpert" hatten und quasi für ihr "stümpern" auch noch
Support gefordert/erwartet hatten, ist das dort jetzt quasi ausgebaut...

ggf. sollte es so funktionieren - wenn Du es Dir zutraust, teste einmal:
die FRITZ.Box_7490-07.57.image im .freetz-dl/fw mit dem entpackten Image aus der
7490er-LabPLUS.zip ersetzen und in menueconfig 7490 und Firmware 07.50+ einstellen
und "Detect the latest firmware version (NEW)" deaktivieren...
und mitdenken und nicht stümpern ;-)

**Comment by fda77** [comment]
Eigentlich geht das auch noch wie gehabt via menuconfig: 
 - 7490 7.57 auswählen
 - Download options > Override firmware options > Firmware source
 - Dort die .image ändern
 - und genau diese Datei auch nach dl/fw/ selbst kopieren

Bei diesen 1000 verschiedenen lab+labor+labplus+inhaus+test+sonstwas von AVM hab ich aber mittlerweile den Überblick verloren was die alle sollen und worin der Vorteil zb von genau dieser "labplus" sein soll.
Gibts dazu ein Changelog von AVM? Oder sind das nur Analsysen der Dateien? Sind die neuer/besser als die release?

**Comment by fda77** [comment]
... und "Detect the latest firmware version" muss aus seins sonst sind die FW-Override Optionen leer

**Reply by aus-sua** [reply]
>  worin der Vorteil zb von genau dieser "labplus" sein soll

Die Labor Plus Versionen sind/waren schnelle Fehlerbehebungen zu (wohl von größeren Providern gemeldeten)
spezifischen Fehlern in einer speziellen Firmwareversion, für die AVM erst einmal keine neue Hauptrevisions-Nr.
vergibt, weil es quasi nur ein Fix dieses (ggf. einzelnen...) Problemes ist.
https://service.avm.de/downloads/LaborPlus/
Diese LaborPlus-Versionen direkt in freetz-NG einzubinden, halte ich für unnötig und auch für zeitverschwendete
Arbeit, weil es ja z.B. den o.g. Weg der Einbindung per menueconfig gibt.
In der letzten Zeit (ab FOS 7.5x) gab es da auch mehrere neue Versionen, quasi ein Fix zum Fix...
Ich glaube, vor FOS 7.5x gab es die letzten LaborPlus-Versionen zur FOS 7.1x...

**Comment by fda77** [comment]
Also sind diese labplus neuer als die 7.57.  Den Link und die Downloads dort kenne ich, aber in den .zip ist nur eine recovery und eine .image. Von einem Changelog ist nichts zu sehen. Daher hab ich diese Versionen ignoriert. Avm sollte mal (u.a.) ein vernünftiges Releasemanagement hinbekommen

**Reply by aus-sua** [reply]
Die "Verbesserungen" sind da in der zip innerhalb der infoLaborPlus.txt beschrieben...

**Comment by fda77** [comment]
Blöd wie ich bin hab ich nur in die 1. Datei https://service.avm.de/downloads/LaborPlus/fritzbox-4060/FRITZ.Box_4060-07.57-108832-LabPLUS.zip reingeschaut. In dieser ist nur .exe und *.image. Wenn man nun auch noch alle anderen (für Geräte die man nicht unbedingt hat) herunterlädt zb für die 7490 findet man wirklich eine  "infoLaborPlus.txt"

Auszug
```
Verbesserungen in FRITZ!OS 07.57-109066-LabPLUS

Telefonie
- Behoben: Abbrüche abgehender Gespräche zu Mobilfunk-Gegenstellen nach 2 Minuten

Internet:
- Behoben: IPv6 Freigabe auf "delegierte Präfixe" wird sporadisch gelöscht 
- Behoben: Problem bei der DNSSEC Validierung in FRITZ!OS

_________________________________________________________


Hinweise zu dieser Beta-Version: 
```

 - Ob dieses Changelog auf alle diese "labplus" zutrifft darf man raten (siehe "beschissenes Releasemanagement")
 - avm sieht die labplus als "beta" an (sowas will ich nicht! Die "Release" sind mir schon zu buggy)
 - wenn avm diese 3 Fehler wirklich in alle 7.57 eingebaut hat müsste eigentlich unverzüglich eine neues "Release" kommen.
   Bei der Wahl zwischen 2 Minuten Telefonaten und riesiger Sicherheitslücke gewinnt meist Asterisk
 - Das Löschen/Verschwinden/Vergessen von IPv6 Freigaben ist auch mal wieder eine Lachnummer

**Reply by Conan179** [reply]
Bei der 6690 ist es nur:
Verbesserungen in FRITZ!OS 07.57-108545

- Internet - Behoben: Problem bei der DNSSEC Validierung in FRITZ!OS

**Comment by schluckspecht** [comment]
Danke für die Tipps. Mit dieser Bugfix Labor klappt nun Anruf zu Mobilfunk wieder ohne Abbrüche, und die Freetz Funktionen sind auch drin.  M.E. müsste für so einen Bug eine echte Release und keine schlecht findbare Labor Plus her, aber AVM scheint langsam in schlechtes Fahrwasser zu geraden. Was ich gelesen habe, haben sie ja auch ab 7.7 oder 7.8 freetz-ng blockiert. Wollte mir ja eigentlich bald die AX zugegen, aber für nen Edelspeedport bezahle ich kein teures Geld mehr. Schade eigentlich.

**Reply by fda77** [reply]
Dieses "blockieren" Threads von Sammy in mehreren Foren hab ich auch schon gesehen. Ob das wirklich so ist müsste jemand überprüfen. Ich hab keine passenden Geräte. Es könnte auch sein dass er einfach eine Firmware für ein anderes Gerät genommen hat. Ausserdem nutzt er ein modifiziertes Freetz

**Comment by fda77** [comment, marked as answer]
Ich find es schon komisch dass AVM seine Nutzer so im Regen stehen lässt. Mit Labor oder Beta bekommt man bei denen überhaupt keinen Support mehr. Und ist AVM der Meinung dass für sie in der GPL Ausnahmen gemacht wurde und sie dann gar keine Sourcen veröffentlichen müssen (das ist natürlich Quatsch).
Daher hatte ich (als ich früher noch einen AVM Router fürs Lan hatte) NIE alpha oder beta darauf installiert. Die Release sind schon verbuggt genug! Wie zB dass man über Telekom Hybrid nicht mehr telefonieren kann, ich glaub das ist seit Fos 7.25 so. Bei der Basisfunktion "Telefonie" hat AVM schon immer viel Mist veranstaltet
